src/Compilers/CSharp/Portable/Binder/Imports.cs

```diff
@@ -576,6 +576,19 @@ private void Validate()
                 alias.Alias.CheckConstraints(semanticDiagnostics);
             }
 
+            foreach (var @using in Usings)
+            {
+                // Check if `using static` directives meet constraints.
+                if (@using.NamespaceOrType.IsType)
+                {
+                    var typeSymbol = (TypeSymbol)@using.NamespaceOrType;
+                    var corLibrary = _compilation.SourceAssembly.CorLibrary;
+                    var conversions = new TypeConversions(corLibrary);
```

Comment:
Do we need to recreate `conversions` for every type or can it be re-used here? Believe it can be re-used as it's the same for every iteration.

Reply:
You're right. There's no reason to believe it will change per-using.

Reply:
Note that basically the same conversions object is created in order to check constraints on using aliases. Any benefit in trying to lazy-load it on the compilation or something and reuse it?

Reply:
> Note that basically the same conversions object is created in order to check constraints on using aliases. Any benefit in trying to lazy-load it on the compilation or something and reuse it?

It is probably safe to use Conversions cached on a compilation in all these places

---
In reply to: [228274243](https://github.com/dotnet/roslyn/pull/30737#discussion_r228274243) [](ancestors = 228274243)